src/coreclr/tools/Common/Compiler/CoreRTNameMangler.cs

```diff
@@ -334,6 +334,11 @@ private string ComputeMangledTypeName(TypeDesc type)
                     {
                         mangledName = GetPrefixMangledTypeName((IPrefixMangledType)type).ToString();
                     }
+                    else if (type is FunctionPointerType pointerType)
+                    {
+                        if (_mangleForCplusPlus) throw new NotImplementedException("Cannot mangle function pointer types when emitting C++");
+                        mangledName = "<fptr>" + pointerType.Signature.ToString();
```

Comment:
`ToString` is not unique enough to guarantee we will not conflict with a different type - this will need to be more complex.

I don't expect we'll need mangling for function pointers until https://github.com/dotnet/runtime/issues/11354 is resolved over in mainstream runtimes.

We end up needing one in latest WPF because the compiler ends up creating a generic instance over a function pointer type internally, but that's a different bug. I've submitted #525 to fix that.